baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/DecisionManager.java

```diff
@@ -266,41 +285,60 @@ public ByteArrayOutputStream getDMN(String customerId, String decisionNameOrId,
      * @return - The deleted version of the Decision.
      */
     @Override
-    public DecisionVersion deleteVersion(String customerId, String decisionNameOrId, long version) {
-        DecisionVersion decisionVersion = findDecisionVersion(customerId, decisionNameOrId, version);
-
-        // Can't delete a version whilst it is current
-        if (DecisionVersionStatus.CURRENT == decisionVersion.getStatus()) {
-            throw new DecisionLifecycleException("It is not valid to delete the 'CURRENT' version of Decision '" + decisionNameOrId + "' for customer id '" + customerId + "'");
-        }
+    public DecisionVersion deleteVersion(String customerId, String decisionName, long version) {
+        DecisionVersion decisionVersion = findDecisionVersion(customerId, decisionName, version);
+        return deleteDecisionVersion(decisionVersion);
+    }
 
-        // Deleting a DecisionVersion is a logical delete. They should still appear in history.
-        decisionVersion.setStatus(DecisionVersionStatus.DELETED);
-        return decisionVersion;
+    /**
+     * Deletes the specified decision fully
+     *
+     * @param decisionId - The id of the Decision to delete.
+     * @return - The deleted decision.
+     */
+    @Override
+    public Decision deleteDecision(String decisionId) {
+        Decision decision = decisionDAO.findByDecisionId(decisionId);
+        return deleteDecision(decision, "<NOT SPECIFIED>", decisionId);
```

Comment:
The `customerId` is available via `Decision.getCustomerId()`, so I think we should use it rather than `<NOT SPECIFIED>` so that we can see when a specific Decision for a particular customer was deleted in the logs.

Reply:
You're right! Fixed